lib/spack/spack/s3_handler.py

```diff
@@ -60,8 +30,24 @@ def _s3_open(url, method="GET"):
     try:
         if method == "GET":
             obj = s3.get_object(Bucket=bucket, Key=key)
-            # NOTE(opadron): Apply workaround here (see above)
-            stream = WrapStream(obj["Body"])
+            # Note: botocore composes StreamingBody(raw_stream), which for old
+            # versions does not inherit anything, making it cumbersome to use
+            # as a stream. The wrapper does the following:
+            # 1. Allow users to set a timeout on read()
+            # 2. Throw when content-length is incorrect.
+            # Right now the benefits of using this class and having and adapter
+            # for old versions etc does not outweigh the benefits.  Instead, we
+            # can use raw_stream directly, which is a
+            # urllib3.response.HTTPResponse object. Unlike
+            # http.client.HTTPResponse it does *not* support peek() out of the
+            # box, so we wrap it in BufferedReader. There's again a caveat, namely
+            # https://github.com/urllib3/urllib3/commit/f0d9ebc41e51c4c4c9990b1eed02d297fd1b20d8
+            # is required for this wrapper to work. Thefore we end up needing
+            # needing urllib3 v1.25.4, which is required from botocore >= 1.19,
```

Comment:
```suggestion
            # urllib3 v1.25.4, which is required from botocore >= 1.19,
```

Nit: `needing` is duplicated